dialog/base_test.go

```diff
@@ -7,23 +7,24 @@ import (
 	"fyne.io/fyne"
 	"fyne.io/fyne/canvas"
 	"fyne.io/fyne/test"
-	"fyne.io/fyne/theme"
 	"fyne.io/fyne/widget"
 )
 
 func TestShowCustom_ApplyTheme(t *testing.T) {
-	test.ApplyTheme(t, theme.DarkTheme())
+	test.NewApp()
+	defer test.NewApp()
+
 	w := test.NewWindow(canvas.NewRectangle(color.Transparent))
-	w.Resize(fyne.NewSize(300, 200))
+	w.Resize(fyne.NewSize(200, 300))
 
 	label := widget.NewLabel("Content")
 	label.Alignment = fyne.TextAlignCenter
 
 	d := NewCustom("Title", "OK", label, w)
 
 	d.Show()
-	test.AssertImageMatches(t, "dialog-custom-dark.png", w.Canvas().Capture())
+	test.AssertImageMatches(t, "dialog-custom-default.png", w.Canvas().Capture())
 
-	test.ApplyTheme(t, theme.LightTheme())
-	test.AssertImageMatches(t, "dialog-custom-light.png", w.Canvas().Capture())
+	test.ApplyTheme(t, test.NewTheme())
```

Comment:
Maybe we should go all in and call this `test.UglyTheme()`

Reply:
> Have you considered how this can be used in the reverse? This would allow developers to define a UI in markup and then convert it into a canvas object. The main challenges I foresee here are mapping resource name to bundled resource, and defining the callbacks eg button.OnTapped, list.Length, list.CreateItem, et al.

No, I didn’t. And, honestly, I am not a fan of defining UI via markup. I prefer to write the code.

Reply:
We might, but not in this PR :).